src/__tests__/hooks.test.tsx

```diff
@@ -0,0 +1,91 @@
+import { render } from "@testing-library/react";
+import React, { Dispatch } from "react";
+
+import { Auth0ProviderAction } from "../actions";
+import * as hooks from "../hooks";
+import { useAuth0, useLog } from "../hooks";
+import { Auth0ProviderContext, AuthenticatedState } from "../types";
+
+const TestComponent: React.FC<{ debug?: boolean; logMessage?: string }> = ({ debug, logMessage }) => {
+  const log = useLog(debug);
+  log(logMessage || "Logging!");
+  return <div />;
+};
+
+describe("hooks", () => {
+  describe("useLog", () => {
+    let origConsole: Pick<typeof window.console, "log" | "warn" | "error">;
+
+    beforeEach(() => {
+      origConsole = {
+        log: window.console.log,
+        warn: window.console.warn,
+        error: window.console.error,
+      };
+
+      window.console.log = jest.fn();
+      window.console.warn = jest.fn();
+      window.console.error = jest.fn();
+    });
+
+    afterEach(() => {
+      window.console.log = origConsole.log;
+      window.console.warn = origConsole.warn;
+      window.console.error = origConsole.error;
+    });
+
+    it("Does not log anything if no arguments are provided", () => {
```

Comment:
I think these tests are fine as is but I wanted to make sure you were aware of https://github.com/testing-library/react-hooks-testing-library which could make things _slightly_ easier.  Probably just removes the `TestComponent` you have above

Reply:
I was _not_ aware.  I'll probably make an issue and correct that in a next iteration.